Add a process for adding and removing subprojects (#28526)
containers/podman: GOVERNANCE.md

Reviewer: Missing a toc entry

Reviewer: unrelated:

```suggestion
* [Contributor Ladder](#contributor-ladder)
```

This document defines the governance of the Podman Project, including its subprojects. It defines the various roles our maintainers fill, how to become a maintainer, and how project-level decisions are made.

# Subprojects

> The Podman project currently consists of the Podman project (the repository containing this file) and two subprojects:

Reviewer: Is this list the source of truth for PCT subprojects? If a new project is added, will this require an update?

Reviewer 2: I agree. There should probably be a document outlining the process for adding or removing subprojects, including specific tasks like updating this list and other related activities. I am not sure if this belongs in the governance documentation, but it would be good to have.

Author: I would say this document would be source of truth; I'll make that explicit.

* [Buildah](https://github.com/containers/buildah)
* [Skopeo](https://github.com/containers/skopeo/)

## Adding Subprojects

> During a Core Maintainers meeting, any Maintainer or Core Maintainer may recommend projects to become a subproject of Podman Container Tools. These projects should have the following characteristics:

Reviewer 3: Does "Podman Container Tools" exist officially in CNCF at this point?

Reviewer 3: That particular verbiage I mean?  I don't recall seeing it.

Author: Yes, that's our official name as a CNCF project

* Are strongly related to Podman Container Tools
* Share the Podman Container Tools mission of delivering a set of tools for building and managing containers on a single node
* Are appropriately licensed and governed or willing to become so
* Are under active development
* Consist of high quality code and designs

> Before submitting an application to the Core Maintainers, the applying project must hold an internal consensus vote of all major contributors to join Podman Container Tools. The Core Maintainers will then review the application, and decide whether or not to accept it. If it is accepted, the Core Maintainers will assign at least one person to assist the new subproject in their integration. The new subproject must map its existing maintainers to the roles defined in the Contributor Ladder below upon being accepted. Appointments of Community Managers and Core Maintainers from the new subproject are subject to approval by the existing Core Maintainers; other roles may be assigned freely initially, but the Contributor Ladder advancement process must be followed after the project is accepted.

Reviewer 3: ```suggestion
Before submitting an application to the Core Maintainers, the project applying must hold an internal consensus vote among all major contributors to join Podman Container Tools. The Core Maintainers will then review the application and decide whether or not to accept it. If it is accepted, the Core Maintainers will assign at least one person to assist with the new subproject's integration. The new subproject must map its existing maintainers to the roles defined in the [Contributor Ladder](#Contributor Ladder) below upon being accepted. Appointments of Community Managers and Core Maintainers from the new subproject are subject to approval by the existing Core Maintainers; other roles may be assigned freely initially, but the Contributor Ladder advancement process must be followed after the project is accepted.
```

Reviewer 3: I think, but don't know for a fact that the link will work.

> Before submitting an application to the Core Maintainers, the project in consideration must hold a consensus vote among all major contributors to join Podman Container Tools. The Core Maintainers will then review the application, and decide whether or not to accept it. If it is accepted, the Core Maintainers will assign at least one person to assist with the new subproject's integration. The new subproject must map its existing maintainers to the roles defined in the [Contributor Ladder](#Contributor Ladder) upon being accepted. Appointments of Community Managers and Core Maintainers from the new subproject are subject to approval by the existing Core Maintainers; other roles may be assigned freely initially, but the Contributor Ladder advancement process must be followed after the project is accepted.

Reviewer: ```suggestion
Before submitting an application to the Core Maintainers, the project in consideration must hold a consensus vote among all major contributors to join Podman Container Tools. The Core Maintainers will then review the application, and decide whether or not to accept it. If it is accepted, the Core Maintainers will assign at least one person to assist with the new subproject's integration. The new subproject must map its existing maintainers to the roles defined in the [Contributor Ladder](#contributor-ladder) upon being accepted. Appointments of Community Managers and Core Maintainers from the new subproject are subject to approval by the existing Core Maintainers; other roles may be assigned freely initially, but the Contributor Ladder advancement process must be followed after the project is accepted.
```

## Removing Subprojects

> In some cases, projects will become inactive or unmaintainable, or wish to separate from Podman Container Tools. Any Core Maintainer may propose removal of a any project on these grounds, and Maintainers can request removal of their own project. The Core Maintainers can confirm this with a majority vote.

Reviewer 4: ```suggestion
In some cases, subprojects will become inactive or unmaintainable, or wish to separate from Podman Container Tools. Any Core Maintainer may propose removal of any subproject on these grounds, and Maintainers can request removal of their own project after a publicly visible majority vote by subproject Maintainers and (e.g., a GitHub issue). The Core Maintainers can confirm this with a majority vote.
```

> In some cases, subprojects will become inactive or unmaintainable, or wish to separate from Podman Container Tools. Any Core Maintainer may propose removal of a subproject on these grounds. Additionally, the Maintainers of any subproject may request removal of their subproject by a publicly visible (e.g. Github issue) majority vote. The Core Maintainers must confirm removals with a majority vote.

Reviewer 3: ```suggestion
In some cases, subprojects will become inactive or unmaintainable, or wish to separate from Podman Container Tools. Any Core Maintainer may propose the removal of a subproject on these grounds. Additionally, the Maintainers of any subproject may request the removal of their subproject by a publicly visible majority vote (e.g., via a GitHub issue). The Core Maintainers must confirm removals with a majority vote.
```
